### PR TITLE
Ensure vendors that are not small businesses cannot bid on auctions > $3500 via API

### PR DESCRIPTION
**Pull request by adelevie**
fixes https://github.com/18F/micropurchase/issues/571

I'm not sure we should have cucumber features for an API story. If I'm wrong, let me know and I'll try to add some.

You'll notice this PR doesn't contain any new application code, just a new request spec (tests the API) and specs on `PlaceBid` (controls all bid placement) to ensure that vendors who aren't small businesses cannot bid on small-business-only auctions.

### CODE REVIEW

**Comment by harrisj** [comment]
We haven't done stories for the other API requests. If we do, I'd suggest we at least tag them with `@api` tags

**Comment by adelevie** [comment]
The `PlaceBid` and request specs seem to me to provide enough assurance. Maybe also add a controller spec to `expect(PlaceBid).to receive(:new)` to describe `BidsController#create`?

File: spec/models/place_bid_spec.rb [hunk 1]
```diff
@@ -31,6 +31,26 @@
...
   end
 
+  context 'when the auction start price is above the micro-purchase threshold' do
+    let(:auction) { create(:auction, :between_micropurchase_and_sat_threshold) }
+
+    context 'and the vendor is not a small business' do
+      let(:current_user) { create(:user, :not_small_business) }
+
+      it 'should reject the bid' do
+        expect { place_bid.perform }.to raise_error(UnauthorizedError)
```
Comment (harrisj) [inline spec/models/place_bid_spec.rb]: Should we add a check that it did not change the bid count?

**Comment by jessieay** [comment]
I like the idea of doing request specs for API tests (rather than cucumber or controller specs). 

The reason for request specs over controller, imo, is that in request specs we can inspect the request body so we can confirm that the JSON matches what we we expect. I think we are already doing this.

**Comment by harrisj** [comment]
I think if we can just add a quick check that the bids count did not change for the error case then I'm happy

**Comment by adelevie** [comment]
Done!

**Comment by harrisj** [comment]
Thank you. Just waiting for Travis to give me the go-ahead and I will merge

**Pull request merged**